platform/source/minimal-printf/mbed_printf_implementation.c

```diff
@@ -625,6 +625,9 @@ int mbed_minimal_formatted_string(char *buffer, size_t length, const char *forma
                     index = next_index;
 
                     mbed_minimal_formatted_string_void_pointer(buffer, length, &result, value, stream);
+                } else if (next == '%') {
+                    index = next_index;
+                    mbed_minimal_formatted_string_signed(buffer, length, &result, next, stream);
```

Comment:
`signed`? Isn't this printing "25"? Surely that should read

    mbed_minimal_formatted_string_character(buffer, length, &result, '%', stream);

Although, just to save space, I wonder if you could cheat and handle this in the overall `else`:

    } else {
        /* Handle `%%` - skip the first `%`, so we print just `%` via the "unrecognized modifier" code */
        if (next == `%`) {
            index++;
        }
       /* Write all characters between format beginning and unrecognized modifier */

I believe that handles what the standard requires - the conversion specifier `%`, with the entire conversion specification being `%%`. Behaviour of "%.5%` is undefined.

Although on the other hand that else loop is clunky by the way. Maybe, instead, the much simpler:

    } else {
        // Unrecognised, or `%%`. Print the `%` that led us in.
        mbed_minimal_formatted_string_character(buffer, length, &result, '%', stream);
        if (next == '%') {
            // Continue printing loop after `%%` 
            index = next_index;
        } 
        // Otherwise we continue the printing loop after the leading `%`, so an
        // unrecognised thing like "Blah = %a" will just come out as "Blah = %a".
    }

Reply:
> Although, just to save space, I wonder if you could cheat and handle this in the overall

Your proposed change increases flash memory usage by 8 bytes.

This [force-push](https://github.com/ARMmbed/mbed-os/compare/8e89d99c1cc774d36869dd3f874c0f8ed448e0f0..ad4eca68ff5dd7f07afeb054faeae78fb14d3f0e) uses `mbed_minimal_formatted_string_character`.

Reply:
The first suggestion, I guess? What about the second - that should be a big reduction.

Reply:
> Although on the other hand that else loop is clunky by the way. Maybe, instead, the much simpler:

This [force-push](https://github.com/ARMmbed/mbed-os/compare/1ebd043e7c870e835f877516e96a1f99bf983f6a..a29776a29c96805d2c6ead99c11c7dd9223c41f4) applies this suggested change for a 44 bytes flash memory saving.

Reply:
Cool. I have one more suggestion that may make it even tighter. Adjust the entire loop to get rid of the `else` on `if (format[index] == '%')`. Make each normal recognised case `continue` instead, then `%` and unrecognised can fall into the standard character print by not `continue`ing.

    for (size_t index = 0; format[index] != '\0'; index++) {
        if (format[index] == '%') {
            //...
            char next = format[next_index];
            if ((next == 'd') || (next == 'i')) {
                //...
                index = next_index;
                continue;
            } else if ((next == 'u') || (next == 'x') || (next == 'X')) {
                //...  Note that putting `index = next_index` last in every case
                // should promote code sharing between each block
                index = next_index;
                continue;
            } else if (next == '%') {
                index = next_index;
            }
        }
        mbed_minimal_formatted_string_character(buffer, length, &result, format[index], stream);
    }

Reply:
That's an optimisation worth trying but I'd rather that we do it in a different PR as we are getting behind on other tasks.